src/controllers/payment.controller.ts

```diff
@@ -25,6 +25,10 @@ export class PaymentController extends BaseController {
     const token: string = this.session.getAccessToken(this.httpContext.request)
     const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
 
+    if (dissolutionSession.applicationPaid === true) {
```

Comment:
This can simply be `if (dissolutionSession.applicationPaid)`.

Though I'm wondering if as soon as the user logs in, they manually navigate to `/payment`, will `applicationPaid` be undefined, therefore bypassing this redirect? Can you test this?

Also I think we mentioned during refinement to check if the Referer header exists in `this.httpContext.request` as an additional check

Reply:
There were some shenanigans while trying to use the referer - the redirect controller wasn't taken into account, and the referer pointed to some other page.

Regarding your test case - I am getting an error page.

Reply:
It must break in the payment.service somewhere. Good the global error page shows in that case so a payment couldn't be made, it's an edge case that shouldn't ever happen.